knr1050Sup/knr1050.db

```diff
@@ -102,90 +81,70 @@ record(bo, "$(P)REMOTE_MODE") {
 
 # ///
-# /// The toggle pump set point will either begin a run or stop a ramp.
-# /// This involves either sending an appropriate ramp AND stop command
-# /// to the device to begin, or the appropriate stop command to end.
-# /// As a result this single set point uses a fanout and link to 
-# /// ensure these commands are correctly sent.
-# ///
-record(bi, "$(P)TOGGLE:PUMP:SP") {
-	field(DESC, "Start or stop the ramp")
-	
-	field(ZNAM, "Stop")
-	field(ONAM, "Start")
-	info(INTEREST, "HIGH")
-}
-
-# ///
-# /// For the given toggle pump set point process the desired record.
+# /// The pump start set point calls both a ramp command as
+# /// well as a forward linked stop mode command. The stop
+# /// mode keeps the last values rather than stopping the
+# /// pump. If the device is in local mode, or the gradients
+# /// are not correctly set, then the pump start set point is
+# /// disabled.
 # ///
-record(calcout, "$(P)_TOG:CALC:PUMP") {
-	field(INPA, "$(P)TOGGLE:PUMP:SP CP")
-	
-	field(CALC, "A+1")
-	
-	field(OUT, "$(P)_TOG:PUMP.SELN PP")
-}
-
-record(fanout, "$(P)_TOG:PUMP") {
-	field(SELM, "Specified")
-	
-	field(LNK1, "$(P)_STOP_PUMP:SP PP")
-	field(LNK2, "$(P)_RAMP:SP PP")
-}
-
-record(bo, "$(P)_STOP_PUMP:SP") {
-	field(DESC, "Stops the ramp")
+record(bo, "$(P)PUMP:STOP:SP") {
+	field(DESC, "Stops the pump")
 	field(DTYP, "stream")
 	field(SCAN, "Passive")
 	
 	field(OUT,  "@knr1050.proto stop_pump $(PORT)")
+	field(ZNAM, "")
+	field(ONAM, "Stop")
 }
 
-record(bo, "$(P)_STOP:KLV:SP") {
-	field(DESC, "Stop mode: keep last values")
-	field(DTYP, "stream")
-
-	field(OUT,  "@knr1050.proto stop_klv $(PORT)")
-}
-
-alias("$(P)_STOP:KLV:SP", "$(P)_STOP:KLV")
+alias("$(P)PUMP:STOP:SP", "$(P)PUMP:STOP")
 
-record(bo, "$(P)_RAMP:SP") {
-	field(DESC, "Starts the ramp")
+record(bo, "$(P)PUMP:START:SP") {
+	field(DESC, "Starts the pump")
 	field(DTYP, "stream")
     field(SCAN, "Passive")
 	
 	field(OUT, "@knr1050.proto ramp($(P),_FLOW:SP:CONV,CON:A:SP,CON:B:SP,CON:C:SP,CON:D:SP) $(PORT)")
+	field(ZNAM, "")
+	field(ONAM, "Start")
 
 	field(FLNK, "$(P)_STOP:KLV:SP PP")
 	
 	field(DISV, "0")
 	field(SDIS, "$(P)DISABLE")
-	field(SIML, "$(P)SIM")
 }
 
-alias("$(P)_RAMP:SP", "$(P)_RAMP")
+alias("$(P)PUMP:START:SP", "$(P)PUMP:START")
+
+record(bo, "$(P)_STOP:KLV:SP") {
+	field(DESC, "Stop mode: keep last values")
+	field(DTYP, "stream")
+
+	field(OUT,  "@knr1050.proto stop_klv $(PORT)")
+}
+
+alias("$(P)_STOP:KLV:SP", "$(P)_STOP:KLV")
 
 ###########################################################
 # Get the device status
 ###########################################################
 
 # ///
-# /// The status is sent to a calc record buffer. The records
-# /// that relate to the status returns are then populated
-# /// using the buffer elements. 
+# /// The status is sent to a calculation record buffer. The 
+# /// records that relate to the status returns are then 
+# /// populated using the buffer elements. 
 # ///
 record(ai, "$(P)GET_STATUS") {
```

Comment:
This should probably be called just `$(P)STATUS:SP`. It should just be a `bo` record if doesn't hold any data.